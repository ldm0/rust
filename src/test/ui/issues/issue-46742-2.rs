unsafe fn add(a: i32, b: i32) -> i32 {
    a + b
}
fn main() {
    let foo = match "+" {
        "+" => add,
        "-" => |a,b| (a - b) as i32,
        _ => unimplemented!(),
    };
    let result: i32 = foo(5, 5); //~ ERROR call to unsafe function
}
