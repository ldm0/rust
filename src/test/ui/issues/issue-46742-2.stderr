error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> $DIR/issue-46742-2.rs:10:23
   |
LL |     let result: i32 = foo(5, 5);
   |                       ^^^^^^^^^ call to unsafe function
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior

error: aborting due to previous error

For more information about this error, try `rustc --explain E0133`.
